### DISCUSSION TITLE
X11 bindings for MacOS

### DISCUSSION THREAD

**Discussion by balewski** [containers/podman#20922] [Q&A]
### Feature request description

I'd like to be able to  use X11 graphics on MacOS while running Ubuntu inside Podman.

This is a demonstrator showing the problem. 

I start image as: 
podman run -it   --user $(id -u):$(id -g)  balewski/ubu22-xeyes:p1

core@dffd00a8735c:/$ xeyes 
Error: Can't open display: 

core@dffd00a8735c:/$ xterm
xterm: Xt error: Can't open display: 
xterm: DISPLAY is not set

The display is not working.

Thanks
Jan

### Additional context

My OS is: Ventura ver 13.6.1
My Podman is podman version 4.7.0
The Dockerfile for the image is: 
https://bitbucket.org/balewski/jannersc/src/master/dockerVaria/ubuntu/Dockerfile.ubu22-xeyes

@lastephey

**Comment by edsantiago** [comment]
X needs a lot of context to work. You need to pass `$DISPLAY` to the container, also the X socket and auth:
```console
$ podman run -v /tmp/.X11-unix:/tmp/.X11-unix:z -v $HOME/.Xauthority:/root/.Xauthority:z -e DISPLAY balewski/.....
```
You almost certainly _do not_ want `--user`, that just confuses things and requires a lot more setup.

You may need `--privileged` for the socket, I'm not sure.

And, this is the extent of my help, because I know nothing about Macs. I didn't even know they had X11.
